src/context/FeedManager.tsx

```diff
@@ -1023,14 +1023,15 @@ export class FeedManager<
     }
   };
 
-  unsubscribe = async () => {
+  unsubscribe = () => {
     const { subscription } = this.state;
     if (!subscription || this.registeredCallbacks.length) {
       return;
     }
 
     try {
-      (await subscription).cancel();
+      // @ts-expect-error
+      subscription?.cancel();
```

Comment:
if you have quick mount/unmounts could this become an issue?

Reply:
We could probably add 
```ts
await subscription;
```
before calling `cancel` on `subscription` object to avoid possible "race conditions" . Object `subscription` seems to be extending Promise class of sorts but the result of this promise-like object does not have `cancel` method available.

Reply:
yea let's do that to be sure